plugins/cloudbuild/src/components/useWorkflowRuns.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useState } from 'react';
+import { useAsyncRetry } from 'react-use';
+import { WorkflowRun } from './WorkflowRunsTable/WorkflowRunsTable';
+import { cloudbuildApiRef } from '../api/CloudbuildApi';
+import { useApi, googleAuthApiRef, errorApiRef } from '@backstage/core';
+import { ActionsListWorkflowRunsForRepoResponseData } from '../api/types';
+
+export function useWorkflowRuns({ projectId }: { projectId: string }) {
+  const api = useApi(cloudbuildApiRef);
+  const auth = useApi(googleAuthApiRef);
+  const errorApi = useApi(errorApiRef);
+
+  const [total, setTotal] = useState(0);
+  const [page, setPage] = useState(0);
+  const [pageSize, setPageSize] = useState(5);
+
+  const { loading, value: runs, retry, error } = useAsyncRetry<
+    WorkflowRun[]
+  >(async () => {
+    const token = await auth.getAccessToken([
+      'https://www.googleapis.com/auth/cloud-platform',
+    ]);
+    return api
```

Comment:
just noting that you could `await` this too to avoid the extra nesting of the `then` below :)